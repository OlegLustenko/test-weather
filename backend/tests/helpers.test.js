const fs = require('fs');
const path = require('path');

const { denormalizeForecast } = require('../helpers');

const normalizedForecast = fs.readFileSync(path.join(__dirname, '/fixtures/normalizedForecast.json'));
const forecast = JSON.parse(normalizedForecast);
const days = 1;
const date = '2017-04-07T15:04:55.535Z';

test('Helper "denormalizeForecast" fits needed format', () => {
  expect(denormalizeForecast(forecast, date, days)).toMatchSnapshot();
});
