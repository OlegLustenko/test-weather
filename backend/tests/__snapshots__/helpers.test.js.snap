// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper "denormalizeForecast" fits needed format 1`] = `
Array [
  Object {
    "date": 2017-04-06T21:00:00.000Z,
    "forecast": Array [
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": null,
        "precipitation": null,
        "temperature": Object {
          "max": null,
          "min": null,
        },
        "weather": null,
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": null,
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": Object {
          "hazardCode": "FW.W",
          "hazardType": "long duration",
          "phenomena": "Red Flag",
          "significance": "Warning",
          "url": "http://forecast.weather.gov/wwamap/wwatxtget.php?cwa=mfl&wwa=Red%20Flag%20Warning",
        },
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": Object {
          "hazardCode": "FW.W",
          "hazardType": "long duration",
          "phenomena": "Red Flag",
          "significance": "Warning",
          "url": "http://forecast.weather.gov/wwamap/wwatxtget.php?cwa=mfl&wwa=Red%20Flag%20Warning",
        },
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
      Object {
        "hazard": Object {
          "hazardCode": "FW.W",
          "hazardType": "long duration",
          "phenomena": "Red Flag",
          "significance": "Warning",
          "url": "http://forecast.weather.gov/wwamap/wwatxtget.php?cwa=mfl&wwa=Red%20Flag%20Warning",
        },
        "icon": "http://www.nws.noaa.gov/weather/images/fcicons/skc.jpg",
        "precipitation": "1",
        "temperature": Object {
          "max": "24",
          "min": "16",
        },
        "weather": Object {
          "additional": Array [
            Object {
              "coverage": "likely",
              "intensity": "light",
              "qualifier": "none",
              "weather-type": "rain showers",
            },
          ],
          "summary": "Rain Showers Likely",
        },
      },
    ],
  },
]
`;
